src/components/TodoApp/TodoApp.tsx

```
@@ -0,0 +1,176 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Todo } from '../../types/types';
+import { TodoFilter } from '../TodoFilter/TodoFilter';
+import { TodoForm } from '../TodoForm/TodoForm';
+import { TodoList } from '../TodoList/TodoList';
+
+const useLocalStorage = (key: string, initialValue: []) => {
+  const [value, setValue] = useState(
+    localStorage.getItem(key) || initialValue,
+  );
+
+  const save = (saveValue: []) => {
+    setValue(saveValue);
+
+    localStorage.setItem('todos', JSON.stringify(saveValue));
+  };
+
+  if (typeof value === 'object') {
+    return [value, save];
+  }
+
+  return [JSON.parse(value), save];
+};
+
+export const TodoApp = () => {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [isToggled, setIsToggled] = useState(false);
+  const { pathname } = useLocation();
+  const activeTodos = todos.filter((todo: Todo) => !todo.completed);
+  const completedTodos = todos.filter((todo: Todo) => todo.completed);
+
+  enum FilterTypes {
+    Active = '/active',
+    Completed = '/completed',
+  }
+  const [userName, setUserName] = useState('');
+
+  async function getUser() {
+    try {
+      const result = await fetch(
+        'https://mate.academy/students-api/users/6057', {
+          method: 'GET',
+          credentials: 'same-origin',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      );
+
+      if (result.status !== 200) {
+        return console.error(result.status);
```

You can throw an error instead